spice_ev/strategies/greedy.py

```diff
@@ -37,66 +37,47 @@ def step(self):
             cs.current_power = 0
 
         for vehicle_id in sorted(self.world_state.vehicles):
+            # get vehicle
             vehicle = self.world_state.vehicles[vehicle_id]
             cs_id = vehicle.connected_charging_station
             if cs_id is None:
                 # not connected
                 continue
-
+            # get connected charging station
             cs = self.world_state.charging_stations[cs_id]
-            gc = self.world_state.grid_connectors[cs.parent]
-
-            charging_stations, avail_bat_power[cs.parent] = load_vehicle(
-                self, cs, gc, vehicle, cs_id, charging_stations, avail_bat_power[cs.parent])
-        # all vehicles loaded
+            gc_id = cs.parent
+            gc = self.world_state.grid_connectors[gc_id]
+            gc_power_left = gc.cur_max_power - gc.get_current_load()
+            delta_soc = vehicle.get_delta_soc()
+            # initialize variables
+            power = 0
+            bat_power_used = False
+            avg_power = 0
+
+            if get_cost(1, gc.cost) <= self.PRICE_THRESHOLD:
+                # low energy price: take max available power from GC (without stationary batteries)
+                power = clamp_power(gc_power_left, vehicle, cs)
+                # charge with power
+                avg_power = vehicle.battery.load(self.interval, power)['avg_power']
+            elif delta_soc > 0:
+                # vehicle needs charging: take max available power (with stationary batteries)
+                power = gc_power_left + avail_bat_power[gc_id]
+                power = clamp_power(power, vehicle, cs)
+                # charge with power
+                avg_power = vehicle.battery.load(
+                    self.interval, max_power=power, target_soc=vehicle.desired_soc)['avg_power']
+                bat_power_used = True
+
+            # add load to charging station
+            charging_stations[cs_id] = gc.add_load(cs_id, avg_power)
+            cs.current_power += avg_power
+
+            if bat_power_used:
+                # check for power from stationary battery
+                avail_bat_power[gc_id] = max(avail_bat_power[gc_id] - avg_power, 0)
+
```

Comment:
balanced.py has these two checks. At least the second check seems useful here to:
```
            # can active charging station bear minimum load?
            assert cs.max_power >= cs.current_power - self.EPS, (
                "{} - {} over maximum load ({} > {})".format(
                    self.current_time, cs_id, cs.current_power, cs.max_power))
            # can grid connector bear load?
            assert gc.cur_max_power >= gc.get_current_load() - self.EPS, (
                "{} - {} over maximum load ({} > {})".format(
                    self.current_time, cs.parent, gc.get_current_load(), gc.cur_max_power))
```